src/AnimationLC.cpp

```diff
@@ -9,26 +9,40 @@
 #include <OgreEntity.h>
 #include <json.h>
 
+#include <iostream>
+
 void AnimationLC::frameRendered(const Ogre::FrameEvent& evt) {
     for (auto anim : animations)
-        if (anim.second->getEnabled())
-            anim.second->addTime(evt.timeSinceLastFrame);
+        if (anim.second->getEnabled()) {
+            if (anim.second->hasEnded()) {
+                anim.second->setEnabled(false);
+            } else {
+                anim.second->addTime(evt.timeSinceLastFrame);
+                std::cout << anim.second->getTimePosition() << '\n';
+            }
+        }
```

Comment:
This could use a negated `if` to reduce indentation:

```suggestion
        if (!anim.second->getEnabled()) continue;
        
        if (anim.second->hasEnded()) {
            anim.second->setEnabled(false);
        } else {
            anim.second->addTime(evt.timeSinceLastFrame);
        }
```

*Also please remove debug `std::cout`s (alongside the `#include <iostream>` at the top) when submitting a pull request.*